src/sentry/api/endpoints/organization_member/details.py

```diff
@@ -198,32 +197,30 @@ def put(
                 # TODO(dcramer): proper error message
                 return Response({"detail": ERR_UNINVITABLE}, status=400)
 
-        if "teams" in result:
-            # dupe code from member_index
-            # ensure listed teams are real teams
-            teams = list(
-                Team.objects.filter(
-                    organization=organization, status=TeamStatus.VISIBLE, slug__in=result["teams"]
-                )
-            )
-
-            if len(set(result["teams"])) != len(teams):
+        # Set the team-role before org-role. If the org-role has elevated permissions
+        # on the teams, the team-roles can be overwritten later
+        if "teamRoles" in result or "teams" in result:
+            try:
+                lock = locks.get(f"org:member:{member.id}", duration=5, name="org_member_details")
+                with TimedRetryPolicy(10)(lock.acquire):
```

Comment:
Wondering if this lock should be inside of `save_team_assignments` instead? Saves needing to take it out at every call site.

Reply:
I don't think we need a lock on this, but I'm not entirely sure about even with this [comment from the author](https://github.com/getsentry/sentry/pull/6054/files/8edbdb181cf898146eda76d46523a21d69ab0ec7#r145798271). 

Just going to move it into `save_team_assignments` for now. Perhaps we can remove it in the future and see if it blows up.